### PR TITLE
feat(filters): add configurable meta filter

### PR DESCRIPTION
**Pull request by Yelinz**
problem in test, when selectivly running tests the settings are overriden and it works. when running the whole test suite the settings arent changed and fails

### CODE REVIEW

File: document_merge_service/api/filters.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        value_new = self.value_transform(value)
+        return super().filter(qs, value_new)
+
+
+class TemplateFilterSet(filters.FilterSet):
+    meta = JSONFieldMultiValueFilter(
+        value_transform=settings.TEMPLATE_META_FILTER_TRANSFORM,
+        lookup_expr=settings.TEMPLATE_META_FILTER_LOOKUP,
+        field_name="meta",
+    )
```
Comment (winged) [inline document_merge_service/api/filters.py:72]: This creates direct references, which will not change when you modify the settings during runtime of the tests. That's probably what you're seeing in CI

**Review by winged — CHANGES_REQUESTED** [review]
Nice work so far, but generally, I'm not sure if this is precisely the way to go. It seems like a rather fragile way to configure configurable meta filters - maybe let's discuss how this should work?

**Comment by winged** [comment]
As discussed in-person: I'd suggest using the `JSONValueFilter` that we wrote in Caluma and is in use in Alexandria as well: https://github.com/projectcaluma/alexandria/blob/main/alexandria/core/filters.py#L12

Sadly, this creates yet another copy of the same filter. I think we should push towards factoring this out into a separate project / library, so it won't need to be copied around too much

**Comment by Yelinz** [comment]
Implemented the filter, will be refactored out at a later date into a separate package

**Comment by czosel** [comment]
Looks like this is still missing some test coverage:
```
document_merge_service/api/filters.py                                       40     23    42%   18-59, 63-70
```

**Comment by Yelinz** [comment]
Forgot to add new test file

File: document_merge_service/api/filters.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+from django_filters.constants import EMPTY_VALUES
+from rest_framework.exceptions import ValidationError
+
+from . import models
+
+
+# TODO: refactor into reusable package later
+class JSONValueFilter(Filter):
+    def filter(self, qs, value):
+        print(value)
```
Comment (czosel) [inline document_merge_service/api/filters.py:14]: This can probably be dropped

**Review by czosel — APPROVED** [review]
Just a little cleanup, otherwise LGTM

File: document_merge_service/api/filters.py (hunk 2 above)
Comment (Yelinz) [reply document_merge_service/api/filters.py]: Done

File: document_merge_service/api/tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        ("foo", HTTP_400_BAD_REQUEST),
+        ("[{foo, no json)", HTTP_400_BAD_REQUEST),
+    ],
+)
+def test_json_value_filter(db, template_factory, admin_client, value, status_code):
+    doc = template_factory(meta={"foo": "bar", "baz": "bla", "int": 23})
+    template_factory(meta={"foo": "baz"})
+    template_factory()
+    url = reverse("template-list")
+    print(url)
```
Comment (czosel) [inline document_merge_service/api/tests/test_filters.py:34]: sorry, another print statement

Comment (Yelinz) [reply document_merge_service/api/tests/test_filters.py]: removed, didnt catch that :(

**Review by czosel — APPROVED** [review]

**Pull request merged**